lib/elixir/src/elixir_erl_compiler.erl

```diff
@@ -56,7 +59,7 @@ compile(Forms, File, Opts) when is_list(Forms), is_list(Opts), is_binary(File) -
           format_warnings(Opts, Warnings),
           {Module, Binary};
 
-        {ok, Module, _Binary, _Warnings} ->
+        {ok, Module, _NotABinary, _Warnings} ->
```

Comment:
Other than this return type not being documented in erlang specs for `compile:noenv_forms/2`, this clause is also not being tested in the elixir test suite (elixir tests are passing after removing this line and the next line). If such test would be useful, it can be added based on an example from https://github.com/elixir-lang/elixir/issues/7839

Reply:
It is triggered if you do things like `ERL_COMPILER_OPTIONS=to_asm` and similar. So we should leave this one as is. :)

Reply:
So this can stay untested? Alright.

Reply:
Right, we just raise after all for clear better errors for people that may be doing the wrong thing. :)